flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/sources/TableSourceUtil.scala

```diff
@@ -116,7 +116,9 @@ object TableSourceUtil {
 
         val (physicalName, idx, tpe) = resolveInputField(name, tableSource)
         // validate that mapped fields are are same type
-        if (fromTypeInfoToLogicalType(tpe) != t) {
+        if (fromTypeInfoToLogicalType(tpe) != t &&
```

Comment:
This looks like a workaround? Maybe we can create another jira to fix this problem in blink?

Reply:
This problem will be fixed after this PR merged: https://github.com/apache/flink/pull/9274

Reply:
I have removed these code in the latest commit.